packages/api/endpoints/collections.js

```diff
@@ -216,24 +230,45 @@ async function put(req, res) {
  * @returns {Promise<Object>} the promise of express response object
  */
 async function del(req, res) {
+  const {
+    collectionsModel = new models.Collection(),
+    collectionPgModel = new CollectionPgModel(),
+    knex = await getKnexClient(),
+    esClient = await Search.es(),
+  } = req.testContext || {};
+
   const { name, version } = req.params;
-  const collectionsModel = new models.Collection();
 
-  const knex = await getKnexClient({ env: process.env });
+  let existingCollection;
   try {
-    await knex.transaction(async (trx) => {
-      await trx(tableNames.collections).where({ name, version }).del();
-      await collectionsModel.delete({ name, version });
-      if (inTestMode()) {
+    existingCollection = await collectionsModel.get({ name, version });
```

Comment:
I really am not in love with this approach. We're looking up the original Dynamo record so that we can restore it in case any of the transactional deletes fail. The problem is that once you have 2 non-PostgreSQL operations in the transaction, one of those could succeed even if the other fails. In the logic below, the Dynamo delete could succeed even if the ES delete subsequently fails. So we need to be able to restore the Dynamo record in that case

I didn't think too hard about alternative approaches, but I couldn't immediately come up with something that felt any better. And ultimately, this complexity goes away once Dynamo goes away.